src/main/java/org/testng/GuiceHelper.java

```diff
@@ -32,16 +39,57 @@ Injector getInjector(IClass iClass) {
 
     List<Module> moduleInstances =
         Lists.newArrayList(getModules(guice, parentInjector, iClass.getRealClass()));
+    List<Module> moduleLookup = Lists.newArrayList(moduleInstances);
+    Module parentModule = getParentModule(context);
+    if (parentModule != null) {
+      moduleInstances.add(parentModule);
+    }
 
-    // Reuse the previous injector, if any
-    Injector injector = context.getInjector(moduleInstances);
+    // Get an injector with the class's modules + any defined parent module installed
+    // Reuse the previous injector, if any, but don't create a child injector as JIT bindings can conflict
+    Injector injector = context.getInjector(moduleLookup);
     if (injector == null) {
-      injector = parentInjector.createChildInjector(moduleInstances);
+      injector = createInjector(context, moduleInstances);
+//      injector = parentInjector.createChildInjector(moduleInstances);
```

Comment:
Trivial observation : Can we pls remove this commented line ?

Reply:
yes, meant to do that, sorry!

Reply:
removed now